internal/seedengine/seedengine.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// Package seedengine provides deterministic key derivation of ECDSA and symmetric keys
+// from a secret seed.
+package seedengine
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"hash"
+	"io"
+
+	"filippo.io/keygen"
+	"golang.org/x/crypto/hkdf"
+)
+
+// SeedEngine provides deterministic key derivation of ECDSA and symmetric keys
+// from a secret seed.
+type SeedEngine struct {
+	curve   func() elliptic.Curve
+	hashFun func() hash.Hash
+	salt    []byte
+
+	podStateSeed []byte
+	historySeed  []byte
+
+	rootCAKey             *ecdsa.PrivateKey
+	transactionSigningKey *ecdsa.PrivateKey
+}
+
+// New creates a new SeedEngine from a secret seed and a salt.
+func New(secretSeed []byte, salt []byte) (*SeedEngine, error) {
```

Comment:
Do we want to require a fixed length for these inputs, too?

Reply:
salt was already checked, I've added secretSeed >= hashLen.